test/subscription_map.hpp

```diff
@@ -337,28 +349,48 @@ class single_subscription_map
     }
 
     // Remove a value at the specified subscription path
-    void erase(MQTT_NS::string_view subscription) {
+    size_t erase(MQTT_NS::string_view subscription) {
```

Comment:
```suggestion
    std::size_t erase(MQTT_NS::string_view subscription) {
```

Could you add `std::` namespace qualifier for `size_t` ?
Please apply the same rule on other part of the code and other file related to trie based wildcard.